src/commands/suspend.ts

```diff
@@ -109,6 +110,24 @@ export const suspendCommand: CommandDef = {
       await user.send(
         'You have been suspended for violating our Code of Conduct. A channel has been created in the server for you to discuss this with the moderation team.'
       );
+      if (config.MONGO_URI) {
+        await userSuspendModel.findOne(
+          { userId: user.id },
+          async (err: Error, data: UserSuspend) => {
+            if (!data) {
+              const newUser = new userSuspendModel({
+                userId: user.id,
+                suspended: true
+              });
+              await newUser.save((err) => console.error(err));
+            } else if (data.suspended) {
+              await message.author.send(
+                `Hello! It looks like ${user} has been suspended previously. You may consider taking further action based on their offence.`
+              );
+            }
+          }
+        );
```

Comment:
this code block mixes callbacks with async/await.

Should be changed to:

```ts
if (config.MONGO_URI) {
  const userSuspend = await userSuspendModel.findOne({ userId: user.id });
  if (userSuspend) {
    await message.author.send(
            `Hello! It looks like ${user} has been suspended previously. You may consider taking further action based on their offence.`
  );
  return;
  }
  const newUser = new userSuspendModel({
    userId: user.id,
    suspended: true
  });
  await newUser.save();
}
```

Reply:
This is done, BTW 🙂